### PR TITLE
doc: release-notes-4.0: Add notes for Wi-Fi

### PR DESCRIPTION
**Pull request by jukkar**
Added release notes for Wi-Fi components.

### CODE REVIEW

File: doc/releases/release-notes-4.0.rst [hunk 1]
```diff
@@ -554,9 +582,31 @@ Drivers and Sensors
...
-  * Added ESP32 driver APSTA support.
+  * Add Wi-Fi Easy Connect (DPP) support.
+  * Add support for Wi-Fi credentials library.
+  * Add enterprise support for station.
+  * Add Wi-Fi snippet support for networking samples.
+  * Add build testing for various Wi-Fi config combinations.
+  * Add regulatory domain support to Wi-Fi shell.
+  * Add WPS support to Wi-Fi shell.
+  * Add 802.11r connect command usage in Wi-Fi shell.
+  * Add currnet PHY rate to hostap status message.
```
Comment (nordicjm) [inline doc/releases/release-notes-4.0.rst:593]: *current

**Comment by jukkar** [comment]
* Fixed the typo

**Comment by jukkar** [comment]
Removed Ethernet part from the release note as there is better one at #81242

**Review by rlubos — APPROVED** [review]

File: doc/releases/release-notes-4.0.rst [hunk 2]
```diff
@@ -554,9 +554,31 @@ Drivers and Sensors
 
 * Wi-Fi
 
-  * Added ESP32-C2 Wi-Fi support.
-  * Added ESP32 driver APSTA support.
+  * Add Wi-Fi Easy Connect (DPP) support.
+  * Add support for Wi-Fi credentials library.
+  * Add enterprise support for station.
```
Comment (krish2718) [inline doc/releases/release-notes-4.0.rst:559]:
```suggestion
  * Add enterprise security support for station.
```

**Review by krish2718 — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**